simple_functorch.py

```diff
@@ -510,6 +515,25 @@ def propagate(dL_doutputs: List[Tensor]):
         )
         return r, saved
 
+    def custom_vmap(self, fn, batch_rule, *args):
+        def call_with_current_dispatcher(fn):
+            def wrapped(d, *args):
+                saved = self.inner
+                try:
+                    self.inner = d
```

Comment:
woooow so spicy

Reply:
I was going to say that you can do this without mutating the dispatcher stack by simply creating a new Autograd dispatcher on the fly, whose inner is d, but then the tapes would not be shared.  This seems... dubious.

Reply:
I guess if we represent the tape with an extra indirection this isn't too hard to do. Probably better and then makes this rule nicely symmetric for how `custom_vjp` is implemented in `Batched`.